### PR TITLE
Convert virt-who cli cases to Pytest

### PR DESCRIPTION
**Pull request by hkx303**
#7749
Converted cases as below:
-  test_positive_deploy_configure_by_id
-  test_positive_deploy_configure_by_script
-  test_positive_debug_option
-  test_positive_interval_option
-  test_positive_hypervisor_id_option
-  test_positive_filter_option
-  test_positive_proxy_option
-  test_positive_rhsm_option
-  test_positive_post_hypervisors

### CODE REVIEW

**Comment by hkx303** [comment]
Test Result:

```
(venv) [hkx303@kuhuang virtwho]$ pytest test_cli_virtwho.py 
==================================== test session starts =====================================
platform linux -- Python 3.7.3, pytest-4.6.3, py-1.8.0, pluggy-0.12.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/hkx303/Documents/CI/robottelo/robottelo
plugins: services-1.3.1, forked-1.0.2, mock-1.10.4, cov-2.7.1, xdist-1.29.0
collecting ... 2020-04-24 03:47:12 - conftest - DEBUG - Collected 9 test cases
2020-04-24 11:47:14 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7f75c92b1828
2020-04-24 11:47:14 - robottelo.ssh - INFO - Connected to [ent-02-vm-08.lab.eng.nay.redhat.com]
2020-04-24 11:47:14 - robottelo.ssh - INFO - >>> rpm -q satellite
2020-04-24 11:47:15 - robottelo.ssh - INFO - <<< stdout
satellite-6.7.0-6.el7sat.noarch

2020-04-24 11:47:15 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7f75c92b1828
2020-04-24 11:47:15 - robottelo.host_info - DEBUG - Host Satellite version: 6.7
2020-04-24 11:47:15 - robottelo.helpers - DEBUG - Generated file bz_cache.json with BZ collect data
collected 9 items                                                                            

test_cli_virtwho.py .........                                                          [100%]

================================ 9 passed in 2924.94 seconds =================================
```

**Comment by swadeley** [comment]
Hello

I see this line removed: 
`    @skip_if_not_set('virtwho')`
but I do not see a corresponding new line with "skipif". You are still reading date from the virtwho section of the robottelo.properties file, so I wonder why thats missing.

**Comment by hkx303** [comment]
> Hello
> 
> I see this line removed:
> ` @skip_if_not_set('virtwho')`
> but I do not see a corresponding new line with "skipif". You are still reading date from the virtwho section of the robottelo.properties file, so I wonder why thats missing.

Hi, Stephen,
Actually, I have considerd this when I removed it. But `@skip_if_not_set` still use `unittest2.SkipTest`, which means maybe we will convert it to another function, and we are not use this in CLI & UI cases, we will add it in another PR if we have the global fixture for sth like skip_if_not_set function.

File: tests/foreman/virtwho/test_cli_virtwho.py [hunk 1]
```diff
@@ -40,45 +39,41 @@
...
-        if self.hypervisor_type == 'libvirt':
-            args['hypervisor-username'] = self.hypervisor_username
-        elif self.hypervisor_type == 'kubevirt':
-            args['kubeconfig'] = self.hypervisor_config_file
-        else:
-            args['hypervisor-username'] = self.hypervisor_username
-            args['hypervisor-password'] = self.hypervisor_password
-        return args
-
+@fixture()
```
Comment (mshriver) [inline tests/foreman/virtwho/test_cli_virtwho.py:42]: Consider module scoping this fixture, as its not changing with the functions.

Comment (mirekdlugosz) [reply tests/foreman/virtwho/test_cli_virtwho.py:42]: Data returned by this fixture is used to create virt-who config entity, and tests remove said entity and assert it doesn't exist by searching for name (generated here). So we need this fixture to return different data for each function, or run tests sequentially.

Comment (mshriver) [reply tests/foreman/virtwho/test_cli_virtwho.py:42]: I see, gen_string is providing some randomization there. Still getting used to the fact that we're executing this parallelization by function across the same target Sat instance, thanks @mirekdlugosz

Comment (hkx303) [reply tests/foreman/virtwho/test_cli_virtwho.py:42]: Sorry for the delayed reply as we had the labor holiday, thanks @mirekdlugosz for the detailed explanation.

**Pull request merged**